File: 프로그래머스/lv1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
s = s.toLowerCase();

        char[] cs = s.toCharArray();
     
        int p_count = 0;
        int y_count = 0;

        for(int i=0; i<cs.length; i++){
            if(cs[i] == 'p'){
                p_count++;
            }
            if(cs[i] == 'y'){
                y_count++;
            }
        }

        if(p_count != y_count){
            answer = false;
        }

        return answer;
    }
}

File: 프로그래머스/1/12916. 문자열 내 p와 y의 개수/문자열 내 p와 y의 개수.js
function solution(s){
    let pCount = 0;
    let yCount = 0;
    
    s = s.toLowerCase();
    for (const char of s) {
        if (char === 'p') pCount++;
        else if (char === 'y') yCount++;
    }


    return pCount === yCount;
}